honeybee_3dm/tojson.py

```diff
@@ -0,0 +1,39 @@
+try:  # import the core honeybee dependencies
+    from honeybee.model import Model
+    from honeybee.config import folders
+except ImportError as e:
+    raise ImportError('\nFailed to import honeybee:\n\t{}'.format(e))
+
+import os
+import json
+
+
+def publish_json(_hb_objs, _name_=None, _folder_=None, indent_=None, abridged_=False, _dump=True):
+
+    # set the component defaults
+    name = _name_ if _name_ is not None else 'unnamed'
+    file_name = '{}.json'.format(name) if len(_hb_objs) > 1 or not \
+        isinstance(_hb_objs[0], Model) else '{}.hbjson'.format(name)
+
+    folder = _folder_ if _folder_ is not None else folders.default_simulation_folder
+    hb_file = os.path.join(folder, file_name)
+    indent = indent_ if indent_ is not None else 0
+    abridged = bool(abridged_)
+
+    # create the dictionary to be written to a JSON file
+    if len(_hb_objs) == 1:  # write a single object into a file if the length is 1
+        try:
+            obj_dict = _hb_objs[0].to_dict(abridged=abridged)
+        except TypeError:  # no abridged option
+            obj_dict = _hb_objs[0].to_dict()
+    else:  # create a dictionary of the objects that are indexed by name
+        obj_dict = {}
+        for obj in _hb_objs:
+            try:
+                obj_dict[obj.identifier] = obj.to_dict(abridged=abridged)
+            except TypeError:  # no abridged option
+                obj_dict[obj.identifier] = obj.to_dict()
+
+    # write the dictionary into a file
+    with open(hb_file, 'w') as fp:
+        json.dump(obj_dict, fp, indent=indent)
```

Comment:
I _feel_ for this library we probably want to have ways to create a `Model` which we can then convert to an energy model or a Radiance folder. Not sure how useful writing the separate faces to a JSON file will be for what we need at this point.

Reply:
To be discussed on the call.

Reply:
I still think we don't need this module. User can use the methods provided in honeybee library to convert the geometries to json or hbjson as needed.

Reply:
I looked [here](https://www.ladybug.tools/honeybee-core/docs/search.html?q=json&check_keywords=yes&area=default). I did not find a method to write hbjson from a Honeybee model. If such a method exists then I would definitely drop this module.

Reply:
Even if it is missing from the honeybee library, we should add it there and re-use it instead of adding it here.

Reply:
Why do we need an entire method to write a model to a JSON? It's literally 2 lines:
```
with open(hb_file, 'w') as fp:
        json.dump(model.to_dict(), fp, indent=indent)
```

Reply:
Looking again at the module here, this whole method is not necessary and has all of this stuff that is not relevant to writing model JSONs. Let's only focus on writing honeybee Models (not any other objects) and just use the two lines I pasted there.

Reply:
I created [This](https://github.com/ladybug-tools/honeybee-core/pull/274) PR. I hope I have understood the comments correctly this time.

Reply:
Created a PR [here](https://github.com/ladybug-tools/honeybee-core/pull/277)